### PR TITLE
Add exemple and clarification about encryption with data source

### PR DESCRIPTION
**Pull request by Gregesp**
#### Summary

Please find the first PR to clarify the possibility of using encryption with data source PostgreSQL or MySQL use by the Mattermost Server.

#### Ticket Link

If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/docs/issues/6803

### CODE REVIEW

**Review by cwarnermm — CHANGES_REQUESTED** [review]
Thank you so much, @Gregesp! I've provided inline feedback that will likely impact the RST table syntax. As you incorporate the changes requested, please review the table formatting as well.

**Review by coltoneshaw — CHANGES_REQUESTED** [review]
Changes as requested by Carrie for wording, and moved to a table for readability

File: source/configure/database-configuration-settings.rst [hunk 1]
```diff
@@ -43,15 +43,34 @@ Data source
...
 | **PostgreSQL databases**                                                                                                                 |
 |                                                                                                                                          |
 | When **Driver Name** is set to ``postgres``, use a connection string in the form of:                                                     |
 | ``postgres://mmuser:password@localhost:5432/mattermost_test?sslmode=disable&connect_timeout=10.``                                        |
+|                                                                                                                                          |
+| **To use TLS with PostgreSQL databases**:                                                                                                |
+|                                                                                                                                          |
+| The parameter to encrypt connection against a PostgreSQL database is ``sslmode`` and it authorize not all mode that you could define     |
```
Comment (cwarnermm) [inline source/configure/database-configuration-settings.rst:53]: "it authorize not all mode that you could define with the CLI" - this sentence isn't easy to parse. 

When ``sslmode`` is used, is CLI functionality limited in some specific way?

File: source/configure/database-configuration-settings.rst [hunk 2]
```diff
@@ -43,15 +43,34 @@ Data source
...
+| with the `CLI (by exemple) <https://www.postgresql.org/docs/current/libpq-ssl.html>`__ like that ``psql "sslmode=value"``                |
+|                                                                                                                                          |
+| The library used to interact with PostgreSQL is `pq <https://pkg.go.dev/github.com/lib/pq>`__ and for this moment you can only use:      |
+|                                                                                                                                          |
+| - ``disable``: No encryption                                                                                                             |
+|                                                                                                                                          |
+| Exemple : ``postgres://mmuser:password@localhost:5432/mattermost_test?sslmode=disable&connect_timeout=10.``                              |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:60]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 3]
```diff
@@ -43,15 +43,34 @@ Data source
...
+|                                                                                                                                          |
+| - ``require`` (default mode if value is omited): Data to be encrypted and i trust the network to connect to the server i think           |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:62]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 4]
```diff
@@ -43,15 +43,34 @@ Data source
...
+|                                                                                                                                          |
+| Exemple : ``postgres://mmuser:password@localhost:5432/mattermost_test?sslmode=require&connect_timeout=10.``                              |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:64]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 5]
```diff
@@ -43,15 +43,34 @@ Data source
...
+|                                                                                                                                          |
+| - ``verify-ca``: Data is encrypted and i want to be sure i connect to a server that i trust                                              |
+|                                                                                                                                          |
+| Exemple : ``postgres://mmuser:password@localhost:5432/mattermost_test?sslmode=verify-ca&connect_timeout=10.``                            |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:68]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 6]
```diff
@@ -72,12 +91,46 @@ Data source
...
+|                                                                                                                                          |
+| - ``tls=false``: To don't use TLS / SSL encryption against MySQL server                                                                  |
+|                                                                                                                                          |
+| Exemple :                                                                                                                                |
+|                                                                                                                                          |
+| ``"<user:pass>@tcp(hostname or IP:3306)/mattermost?charset=utf8mb4,utf8&writeTimeout=30s&tls=false"``                                    |
+|                                                                                                                                          |
+| - ``tls=true``: To use TLS / SSL encryption against MySQL server.                                                                        |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:117]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 7]
```diff
@@ -72,12 +91,46 @@ Data source
...
+|                                                                                                                                          |
+| Exemple :                                                                                                                                |
+|                                                                                                                                          |
+| ``"<user:pass>@tcp(hostname or IP:3306)/mattermost?charset=utf8mb4,utf8&writeTimeout=30s&tls=true"``                                     |
+|                                                                                                                                          |
+| - ``tls=skip-verify``: To use TLS / SSL encryption with self-signed certificates                                                         |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:123]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 8]
```diff
@@ -72,12 +91,46 @@ Data source
...
+|                                                                                                                                          |
+| Exemple :                                                                                                                                |
+|                                                                                                                                          |
+| ``"<user:pass>@tcp(hostname or IP:3306)/mattermost?charset=utf8mb4,utf8&writeTimeout=30s&tls=skip-verify"``                              |
+|                                                                                                                                          |
+| - ``tls=preferred``: To use TLS / SSL encryption if server advertise with a possible fallback unencrypted if it not advertised           |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:129]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst (hunk 1 above)
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:53]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 9]
```diff
@@ -43,15 +43,34 @@ Data source
...
+|                                                                                                                                          |
+| Exemple : ``postgres://mmuser:password@localhost:5432/mattermost_test?sslmode=disable&connect_timeout=10.``                              |
+|                                                                                                                                          |
+| - ``require`` (default mode if value is omited): Data to be encrypted and i trust the network to connect to the server i think           |
+|                                                                                                                                          |
+| Exemple : ``postgres://mmuser:password@localhost:5432/mattermost_test?sslmode=require&connect_timeout=10.``                              |
+|                                                                                                                                          |
+| - ``verify-ca``: Data is encrypted and i want to be sure i connect to a server that i trust                                              |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:66]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 10]
```diff
@@ -72,12 +91,46 @@ Data source
...
+|                                                                                                                                          |
+| Exemple :                                                                                                                                |
+|                                                                                                                                          |
+| ``"<user:pass>@tcp(hostname or IP:3306)/mattermost?charset=utf8mb4,utf8&writeTimeout=30s&tls=skip-verify"``                              |
+|                                                                                                                                          |
+| - ``tls=preferred``: To use TLS / SSL encryption if server advertise with a possible fallback unencrypted if it not advertised           |
+|                                                                                                                                          |
+| Exemple :                                                                                                                                |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:131]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst [hunk 11]
```diff
@@ -72,12 +91,46 @@ Data source
...
+|                                                                                                                                          |
+| ``"<user:pass>@tcp(hostname or IP:3306)/mattermost?charset=utf8mb4,utf8&writeTimeout=30s&tls=true"``                                     |
+|                                                                                                                                          |
+| - ``tls=skip-verify``: To use TLS / SSL encryption with self-signed certificates                                                         |
+|                                                                                                                                          |
+| Exemple :                                                                                                                                |
```
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:125]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

File: source/configure/database-configuration-settings.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (Gregesp) [reply source/configure/database-configuration-settings.rst:131]: Application of the recommendations as suggested by @cwarnermm and @coltoneshaw

**Comment by cwarnermm** [comment]
@Gregesp - Are you open to signing the CLA and merging the latest master branch into your PR?

**Comment by Gregesp** [comment]
@cwarnermm : I approved the CLA and Merged master Branch into my Pull Request

**Comment by cwarnermm** [comment]
Thanks so much, @Gregesp! Next steps for this PR include a technical review.

**Review by cwarnermm — APPROVED** [review]

**Review by coltoneshaw — COMMENTED** [review]
Looking at the PR, the structure/documentation is excellent. Thanks @Gregesp!!

My only feedback so far would be that it's not a simple plug-and-play here, as noted by the `your db admin must configure it`. I played around with enabling it, and funny enough, most everything I found suggested disabling it.

I'm curious if it's the role of Mattermost to document the various connectivity modes to a database, or if the better approach is to just link to the documentation from Postgres. 

If we want to include it as it is, then I'm good for it because it's a nice port of the docs within MySQL / postgres.

**Comment by cwarnermm** [comment]
@Gregesp - Given that master branch changes hourly/daily, would you be open to allowing Mattermost staff to merge latest master into this PR? If you are, it'll speed up our ability to finalize this PR and merge it into our codebase.

**Comment by Gregesp** [comment]
Hi @cwarnermm,
No problem, you can take in hand this PR 👍

**Comment by cwarnermm** [comment]
@Gregesp: In the generated preview, there's additional spacing between lines that make reading the text challenging:
image

Would you be open to removing those extra lines across both tables?

**Comment by Gregesp** [comment]
Hello @cwarnermm,
Yes you could remove the extra lines.
When i start my PR, i used the GitHub preview, i don't build the python sphinx-doc locally (in fact, i should have), and i do add this extra lines.

Gregesp

**Comment by cwarnermm** [comment]
@Gregesp - I'm continuing to have trouble keeping this fork updated against the master branch. Please sync this fork with master when you have a chance. My plan is to merge in this change as-is, and then touch up the spacing issues in a subsequent PR. Thank you for your patience with this!

**Pull request merged**